app/src/main/java/com/example/restopass/main/ui/settings/communications/CommunicationsFragment.kt

```diff
@@ -0,0 +1,173 @@
+import com.example.restopass.main.ui.settings.personalInfo.PersonalInfoViewModel
+import com.example.restopass.utils.AlertDialogUtils
+import com.google.android.material.textfield.TextInputLayout
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.fragment_communications.*
+import kotlinx.coroutines.*
+import timber.log.Timber
+
+class CommunicationsFragment : Fragment(), SecondaryEmailAdapter.SecondaryEmailListener {
+
+    private lateinit var viewModel: PersonalInfoViewModel
+
+    private lateinit var secondaryEmailsRecyclerView: RecyclerView
+    private lateinit var emailAdapter: SecondaryEmailAdapter
+
+    private val emailRegexes = ValidationFactory.emailValidations
+
+    var job = Job()
+    var coroutineScope = CoroutineScope(job + Dispatchers.Main)
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+
+        return inflater.inflate(R.layout.fragment_communications, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewModel = ViewModelProvider(requireActivity()).get(PersonalInfoViewModel::class.java)
+        emailAdapter =
+            SecondaryEmailAdapter(
+                this,
+                viewModel
+            )
+        secondaryEmailsRecyclerView = secondaryEmailsRecyclerContainer.apply {
+            layoutManager = LinearLayoutManager(this.context)
+            adapter = emailAdapter
+        }
+
+        addSecondaryEmailButton.setOnClickListener {
+            onAddEmailClick(secondaryEmailInputLayout.editText?.text.toString())
+        }
+
+        (activity as MainActivity).topAppBar?.apply {
+            setTitle(R.string.communications_settings)
+            visibility = View.VISIBLE
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+
+        communicationsLoader.visibility = View.VISIBLE
+        communicationsSection.visibility = View.GONE
+        coroutineScope.launch {
+            try {
+                viewModel.get()
+
+                communicationsLoader.visibility = View.GONE
+                communicationsSection.visibility = View.VISIBLE
+            } catch (e: Exception) {
+                if (isActive) {
+                    Timber.e(e)
+                    communicationsLoader.visibility = View.GONE
+                    AlertDialogUtils.buildAlertDialog(e, layoutInflater, communicationsContainer, view).show()
+                }
+            }
+        }
+
+
+    }
+
+    private fun isValid(validations: List<Validation>, layout: TextInputLayout) : Boolean {
+        validations.find {
+            !it.regex.matches(layout.editText?.text.toString())
+        }?.let {
+            layout.error = it.errorMessage
+            return false
+        }
+
+        layout.error = null
```

Comment:
porque asignas null aca

Reply:
básicamente porque si primero tuvo un error hago un layout.error = it.errorMessage. Cuando lo corrige, tengo que sacar ese mensaje, así que lo remuevo de esa manera

Reply:
ahh okey